<launch>

	<!-- 사용자 설정 매개변수 시작 -->
	<arg name="lvx_file_path" default="livox_test.lvx"/> <!-- lvx 파일 경로 설정 -->
	<arg name="bd_list" default="100000000000000"/> <!-- 기기 시리얼 번호 설정 -->
	<arg name="xfer_format" default="0"/> <!-- 데이터 전송 형식 설정 -->
	<arg name="multi_topic" default="0"/> <!-- 다중 토픽 사용 여부 설정 -->
	<arg name="data_src" default="0"/> <!-- 데이터 소스 설정 -->
	<arg name="publish_freq" default="10.0"/> <!-- 데이터 발행 주기 설정 -->
	<arg name="output_type" default="0"/> <!-- 출력 데이터 형식 설정 -->
	<arg name="rviz_enable" default="true"/> <!-- RViz 사용 여부 설정 -->
	<arg name="rosbag_enable" default="false"/> <!-- Rosbag 사용 여부 설정 -->
	<arg name="cmdline_arg" default="$(arg bd_list)"/> <!-- 명령줄 인수 설정 -->
	<arg name="msg_frame_id" default="livox_frame"/> <!-- 메시지 프레임 ID 설정 -->
	<arg name="lidar_bag" default="true"/> <!-- 라이다 데이터 기록 여부 설정 -->
	<arg name="imu_bag" default="true"/> <!-- IMU 데이터 기록 여부 설정 -->
	<!-- 사용자 설정 매개변수 끝 -->

	<!-- ROS 매개변수 설정 -->
	<param name="xfer_format" value="$(arg xfer_format)"/> <!-- 전송 형식 설정 -->
	<param name="multi_topic" value="$(arg multi_topic)"/> <!-- 다중 토픽 설정 -->
	<param name="data_src" value="$(arg data_src)"/> <!-- 데이터 소스 설정 -->
	<param name="publish_freq" type="double" value="$(arg publish_freq)"/> <!-- 발행 주기 설정 -->
	<param name="output_data_type" value="$(arg output_type)"/> <!-- 출력 데이터 형식 설정 -->
	<param name="cmdline_str" type="string" value="$(arg bd_list)"/> <!-- 명령줄 문자열 설정 -->
	<param name="cmdline_file_path" type="string" value="$(arg lvx_file_path)"/> <!-- 명령줄 파일 경로 설정 -->
	<param name="user_config_path" type="string" value="$(find livox_ros_driver2)/config/HAP_config.json"/> <!-- 사용자 설정 파일 경로 설정 -->
	<param name="frame_id" type="string" value="$(arg msg_frame_id)"/> <!-- 프레임 ID 설정 -->
	<param name="enable_lidar_bag" type="bool" value="$(arg lidar_bag)"/> <!-- 라이다 데이터 기록 설정 -->
	<param name="enable_imu_bag" type="bool" value="$(arg imu_bag)"/> <!-- IMU 데이터 기록 설정 -->

	<!-- Livox Lidar 노드 실행 -->
	<node name="livox_lidar_publisher2" pkg="livox_ros_driver2"
	      type="livox_ros_driver2_node" required="true"
	      output="screen" args="$(arg cmdline_arg)"/>

	<!-- RViz 사용 설정 그룹 -->
	<group if="$(arg rviz_enable)">
		<node name="livox_rviz" pkg="rviz" type="rviz" respawn="true"
				args="-d $(find livox_ros_driver2)/config/display_point_cloud_ROS1.rviz"/> <!-- RViz 노드 실행 -->
    </group>

	<!-- rosbag 사용 설정 그룹 -->
	<group if="$(arg rosbag_enable)">
    	<node pkg="rosbag" type="record" name="record" output="screen"
          		args="-a"/> <!-- 모든 토픽을 기록하는 rosbag 노드 실행 -->
    </group>

</launch>
